### DISCUSSION TITLE
Deal with increment parallelism

### DISCUSSION THREAD

**Discussion by l-you** [doctrine/doctrine2#10082] [Support Questions] [answered]
Suppose we have e-commerce website with hot sales. There often can be a case where product stock is decremented in parallel requests.
Solution can be to use mysql directly `UPDATE product SET stock=stock-1`.
But it has drawbacks. For example task is  "Create order and reduce stock". I reduce stock with direct mysql query, but during 
`->flush()` exception appears and order not saved. So I have reduced stock and no order.
Is there a way `UPDATE product SET stock=stock-1` to be persisted, so it will execute with other updates when `->flush()` is called?

**Comment by greg0ire** [comment, marked as answer]
From [the docs](https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/transactions-and-concurrency.html#approach-2-explicitly):

> Explicit transaction demarcation is required when you want to include custom DBAL operations in a unit of work

I think that's what you want to do.

**Reply by l-you** [reply]
This approach requires a query to be executed between `$em->getConnection()->beginTransaction();` and `$em->getConnection()->commit();`.
I there a built-in way to create query outside and delay it until `->beginTransaction()`, or should I do it manually?
It would be good to have something like `->persist()` / `->detach()` for DQL/SQL queries

**Reply by greg0ire** [reply]
Not as far as I know.
